File: Javascript/Q53-Maximum-Subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var maxSubArray = function(nums) {
    var maxSum=-2147483648;
       var sum=0;
       for(var i of nums){
           sum+=i;
           maxSum=Math.max(maxSum,sum);
           if(sum<0){
               sum=0;
           }
       }
       return maxSum;
};

File: STRIVER-SDE-SHEET/Arrays-part-1/Q53-Maximum-Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum=Integer.MIN_VALUE;
        int sum=0;
        for(int i:nums){
            sum+=i;
            maxSum=Math.max(maxSum,sum);
            if(sum<0){
                sum=0;
            }
        }
        return maxSum;
    }
}